common/services/prismic/opening-times.js

```diff
@@ -108,211 +94,185 @@ function exceptionalOpeningPeriods(dates: OverrideDate[]): ExceptionalPeriod[] {
-                    'YYYY-MM-DD'
-                  )
-                );
-              }
+export function backfillExceptionalVenueDays(
+  venue: Venue,
+  allVenueExceptionalPeriods?: { type: OverrideType, dates: Moment[] }[]
+): ExceptionalOpeningHoursDay[][] {
+  const groupedExceptionalDays = groupExceptionalVenueDays(
+    getExceptionalVenueDays(venue)
+  );
+  return allVenueExceptionalPeriods
+    ? allVenueExceptionalPeriods
+        .map(period => {
+          const sortedDates = period.dates.sort((a, b) => {
+            return a.diff(b, 'days');
+          });
+          const type = period.type || 'other';
+          const days = sortedDates.map(date => {
+            const matchingVenueGroup = groupedExceptionalDays.find(group => {
+              return group.find(day => day.overrideDate.isSame(date, 'day'));
             });
-          const changes = identifyChanges(override, place, exceptionalDay);
-          const openingHours =
-            override ||
-            place.openingHours.regular.find(
-              item => item.dayOfWeek === exceptionalDay
-            );
-          return {
-            exceptionalDate,
-            exceptionalDay,
-            id: place.id,
-            name: place.name,
-            order: place.order,
-            openingHours,
-            opensChanged: changes.opensHasChanged,
-            closesChanged: changes.closesHasChanged,
-          };
-        });
-        acc.push(overrides);
-
-        return acc;
-      }, [])
-    )
-    .sort((a, b) => {
-      return a.order - b.order;
-    });
+            const matchingDay =
+              matchingVenueGroup &&
+              matchingVenueGroup.find(day =>
+                day.overrideDate.isSame(date, 'day')
+              );
+            const backfillDay = exceptionalFromRegular(venue, date, type);
+            return matchingDay || backfillDay;
+          });
+          return days;
+        })
+        .filter(Boolean)
+    : [];
 }
 
-function upcomingExceptionalOpeningPeriods(
-  periods: ExceptionalPeriod[],
-  daysInAdvance: number = 15
-) {
-  const exceptionalPeriods =
-    periods &&
-    periods.filter(period => {
-      const displayPeriodStart = london().subtract(1, 'day');
-      const displayPeriodEnd = london().add(daysInAdvance, 'day');
-      return (
-        period.dates[0].overrideDate
-          .clone()
-          .isBetween(displayPeriodStart, displayPeriodEnd) ||
-        period.dates[period.dates.length - 1].overrideDate
-          .clone()
-          .isBetween(displayPeriodStart, displayPeriodEnd)
-      );
-    });
-
-  return exceptionalPeriods.length > 0 ? exceptionalPeriods : null;
+export function groupConsecutiveDays(
+  openingHoursPeriods: ExceptionalOpeningHoursDay[][]
+): ExceptionalOpeningHoursDay[][] {
+  const flattenedArray = openingHoursPeriods.reduce(
+    (acc, curr) => acc.concat(curr),
+    []
+  );
+  return flattenedArray.length > 0
```

Comment:
This conditional doesn't make sense, as if the `length` is 0, running the `sort / reduce` would just return an array any way?